src/tree/link.rs

```diff
@@ -61,21 +61,51 @@ impl<'a> LinkLocation<'a> {
         let mut link_str = link.as_ref();
 
         // Check for direct URLs or anchor links
-        if is_url(link_str) || link_str.starts_with('#') {
+        // TODO: parse local links into LinkLocation::Page
+        // Known bug: single "/" parsed into Url instead of Page
+        if is_url(link_str) || link_str.starts_with('#') || link_str.starts_with("/") {
             return LinkLocation::Url(link);
         }
 
-        // Check for local links starting with '/'
-        if link_str.starts_with('/') {
-            link_str = &link_str[1..];
-        }
+        // // Check for local links starting with '/'
+        // if link_str.starts_with('/') {
+        //     link_str = &link_str[1..];
+        // }
+
+        // Take only the first segment for page
+        link_str = link_str.split('#').collect::<Vec<&str>>()[0]
+            .split('/')
+            .collect::<Vec<&str>>()[0];
 
         match PageRef::parse(link_str) {
-            Err(_) => LinkLocation::Url(link),
+            Err(_) => LinkLocation::Url(Cow::Owned(link_str.to_owned())),
```

Comment:
Why can't you use `Cow::Borrowed()` here? I think that's what it did before.

Reply:
Compiler throws error
```cannot return value referencing function parameter `link`
returns a value referencing data owned by the current function```

Transferring ownership seems to be the easiest way to go about it

Reply:
Ah one branch allocates new. Okay sure